usagov/usagov-2021
# USAGOV-1907: Add Data Layer Push for Selection

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1907

## Description


This Pr adds a data-layer push for GTM.  
1 - for a successful section on the benefits search page 
2 - if there is an error in the selection  

Also adding a `data-analytics` to target the error message 

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



1. Review files changed and that data.layer push was added correctly

### Change Requirements



### Validation Steps

- visit `benefit-finder` and open up the console and type in `dataLayer` in the console
- click the apply selection with none selected and verify that you see `category_apply_error`
- click on one option and then verify that you see `category_apply_success` 

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
